ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1409,6 +1412,29 @@ public Collection<AmbryPartition> getPartitions() {
       return new ArrayList<>(partitionMap.values());
     }
 
+    /**
+     *
+     * @return boolean regarding whether we want to filter partitions
+     */
+    public boolean isPartitionFilteringEnabled() {
+      return clusterMapConfig.clusterMapPartitionFilteringEnabled;
+    }
+
+    /**
+     *
+     * @param partitionID
+     * @return boolean regarding whether we can write to it
+     */
+    @Override
+    public boolean isValidPartition(String partitionID) {
+      String resource = getResourceForPartitionInLocalDc(partitionID).iterator().next();
+      ConcurrentHashMap<String, String> res = dcToResourceNameToTag.get(clusterMapConfig.clusterMapDatacenterName);
+      String tag = dcToResourceNameToTag.get(clusterMapConfig.clusterMapDatacenterName).get(resource);
+      ResourceProperty resourceProperty  =
+          dcToTagToResourceProperty.get(clusterMapConfig.clusterMapDatacenterName).get(tag);
```

Comment:
Let's put this logic in a try catch statement. Just to be sure that we don't run into any synchronization error. If there any exception fetching out those metadata, partition should be invalid.